test/parallel/test-stream-pipeline.js

```diff
@@ -1616,3 +1616,20 @@ const tsp = require('timers/promises');
   dup.push(null);
   dup.read();
 }
+
+{
+  let res = '';
+  const writable = new Writable({
+    write(chunk, enc, cb) {
+      res += chunk;
+      cb();
+    }
+  });
+  pipelinep(async function*() {
+    yield 'hello';
```

Comment:
Does it actually works with `async` wait between the `yield`?

```js
{
  let res = '';
  const writable = new Writable({
    write(chunk, enc, cb) {
      res += chunk;
      cb();
    }
  });
  pipelinep(async function*() {
    yield 'hello';
    await setTimeout(100)
    yield 'world';
  }, writable, { end: false }).then(common.mustCall(() => {
    assert.strictEqual(res, 'helloworld');
    assert.strictEqual(writable.closed, false);
  }));
}
```

Reply:
Passes
<img width="1069" alt="Screenshot 2023-04-25 at 6 46 48 PM" src="https://user-images.githubusercontent.com/20257253/234288360-ffa4d578-6b60-4a70-8686-27f5ffba5ff6.png">

would you prefer the timer version of the test to be there?

Reply:
If you do not like to use `timers/promises`, `await Promise.resolve()` should be the same.

Reply:
Updated PTAL :-)

Reply:
Thanks for verifying it works.